make dev-env on Intel Mac broken because of --load flag

- There have been some discussions about this on community-meeting and slack
- `make dev-env` is broken on Intel macOS Monterey with docker context working fine and even `kind create cluster` working
- The error reported is `unknown flag: --load`
- Contributors with Intel Mac depend on this
- Not sure if this is a problem on M1 Mac or Linux

```
k8s.io/apiserver/pkg/server/healthz
k8s.io/ingress-nginx/pkg/metrics
k8s.io/ingress-nginx/cmd/nginx
+ echo 'Building k8s.io/ingress-nginx/cmd/dbg'
Building k8s.io/ingress-nginx/cmd/dbg
+ go build -v -trimpath '-ldflags=-buildid= -w -s     -X k8s.io/ingress-nginx/version.RELEASE=1.0.0-dev     -X k8s.io/ingress-nginx/version.COMMIT=git-f0490cbfb     -X k8s.io/ingress-nginx/version.REPO=git@github.com:longwuyuan/ingress-nginx.git' -o rootfs/bin/amd64/dbg k8s.io/ingress-nginx/cmd/dbg
go: downloading github.com/spf13/cobra v1.5.0
github.com/spf13/cobra
k8s.io/ingress-nginx/cmd/dbg
+ echo 'Building k8s.io/ingress-nginx/cmd/waitshutdown'
Building k8s.io/ingress-nginx/cmd/waitshutdown
+ go build -v -trimpath '-ldflags=-buildid= -w -s     -X k8s.io/ingress-nginx/version.RELEASE=1.0.0-dev     -X k8s.io/ingress-nginx/version.COMMIT=git-f0490cbfb     -X k8s.io/ingress-nginx/version.REPO=git@github.com:longwuyuan/ingress-nginx.git' -o rootfs/bin/amd64/wait-shutdown k8s.io/ingress-nginx/cmd/waitshutdown
k8s.io/ingress-nginx/cmd/waitshutdown
removing old image gcr.io/k8s-staging-ingress-nginx/controller:1.0.0-dev
Error: No such image: gcr.io/k8s-staging-ingress-nginx/controller:1.0.0-dev
Building docker image (amd64)...
unknown flag: --load
See 'docker build --help'.
make[1]: *** [image] Error 125
make: *** [dev-env] Error 2
m@mypc [~/Documents/github/longwuyuan/ingress-nginx] make-broken
% make clean && make dev-env
```

- docker and kind are working 
```
% docker ps
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES

% docker images
REPOSITORY                                      TAG       IMAGE ID       CREATED       SIZE
gcr.io/k8s-staging-ingress-nginx/nginx          <none>    8bea7fc9758e   4 days ago    161MB
registry.k8s.io/ingress-nginx/e2e-test-runner   <none>    8b1253c38468   4 weeks ago   1.17GB
kindest/node                                    v1.24.2   d8d5552a11d9   4 weeks ago   912MB

% kind get clusters  
No kind clusters found.

% kind create cluster
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.24.0) 🖼 
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Not sure what to do next? 😅  Check out https://kind.sigs.k8s.io/docs/user/quick-start/

% kind get clusters
kind
% k cluster-info
Kubernetes control plane is running at https://127.0.0.1:63495
CoreDNS is running at https://127.0.0.1:63495/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.

% k get ns
NAME                 STATUS   AGE
default              Active   26s
kube-node-lease      Active   28s
kube-public          Active   28s
kube-system          Active   28s
local-path-storage   Active   21s
% 
```

/triage accepted
/area stablization
/priority important-soon
/assign @strongjz @tao12345666333

--- a/Makefile
+++ b/Makefile
@@ -76,7 +76,6 @@ image: clean-image ## Build image for a particular arch.
 	docker build \
 		${PLATFORM_FLAG} ${PLATFORM} \
 		--no-cache \
-		$(MAC_DOCKER_FLAGS) \
 		--pull \
 		--build-arg BASE_IMAGE="$(BASE_IMAGE)" \
 		--build-arg VERSION="$(TAG)" \
@@ -94,7 +93,6 @@ image-chroot: clean-chroot-image ## Build image for a particular arch.
 	echo "Building docker image ($(ARCH))..."
 	docker build \
 		--no-cache \
-		$(MAC_DOCKER_FLAGS) \
 		--pull \
 		--build-arg BASE_IMAGE="$(BASE_IMAGE)" \
 		--build-arg VERSION="$(TAG)" \
@@ -210,7 +208,6 @@ dev-env-stop: ## Deletes local Kubernetes cluster created by kind.
 live-docs: ## Build and launch a local copy of the documentation website in http://localhost:8000
 	@docker build ${PLATFORM_FLAG} ${PLATFORM} \
                   		--no-cache \
-                  		$(MAC_DOCKER_FLAGS) \
                   		 -t ingress-nginx-docs .github/actions/mkdocs
 	@docker run ${PLATFORM_FLAG} ${PLATFORM} --rm -it \
 		-p 8000:8000 \
@@ -250,6 +247,7 @@ release: ensure-buildx clean
 
 	docker buildx build \
 		--no-cache \
+		$(MAC_DOCKER_FLAGS) \
 		--push \
 		--pull \
 		--progress plain \
@@ -262,6 +260,7 @@ release: ensure-buildx clean
 
 	docker buildx build \
 		--no-cache \
+		$(MAC_DOCKER_FLAGS) \
 		--push \
 		--pull \
 		--progress plain \

